### ISSUE TITLE
Feature-Request: Actions as buttons

### ISSUE THREAD

**Issue by bernhardh** [laravel/nova-issues#451]
It would be great, to get a similar option to add buttons like backpack has (https://laravel-backpack.readme.io/docs/crud-buttons).

There are several usecases:

**Open ressource on frontend website**
If you have a blogpost (or product, or anything else) you want to directly jump to the frontend url of the post

**Export ressource as CSV/PDF/etc**
If you have an invoice ressource, you might directly download it as PDF.

**Any action you can think of**
Basically any Nova Action you can access over the dropdown, but as a one click shorthand

**Comment by davidhemphill** [comment]
You can do most of this with the newly-released `asHtml` option for computed Text fields: https://nova.laravel.com/docs/1.0/resources/fields.html#computed-fields

We're coming up with a way to add buttons to the index and detail pages directly, but for now the suggestion is to use a computed field.

**Comment by bernhardh** [comment]
Thank you for this hint and its a good workaround for the moment for per-row action.

But whats about adding buttons to index table-wide. I know I can do it over actions in the dropdown, but from a usability perspective, often used actions should be direct accessable.

**Comment by davidhemphill** [comment]
As I mentioned we're coming up with a way to do just that. :-)

**Comment by davidhemphill** [comment]
This has been added and will be included in an upcoming release.

**Comment by bernhardh** [comment]
Thank you for your work! I saw "index and detail headers and toolbars that may be customized for your own Nova installation" in v1.1.0 relase notes, but can't find any documentation how to implement it?

**Comment by davidhemphill** [comment]
We're going to document it later, but basically you just need to copy the contents of those components and then register your own in a Tool.

**Comment by loveyuxiao** [comment]
From which document can find this function？

**Comment by marconett** [comment]
@davidhemphill Please be more specific. Trying to do this as well.

**Comment by davidhemphill** [comment]
Create a new javascript file in your Mix configuration. In your `NovaServiceProvider`, register it with Nova as a script:

```php
public function boot()
{
        Nova::serving(function (ServingNova $event) {
            Nova::script('custom-resource-table-row', asset('custom-resource-table-row.js'));
        });
}
```

Alternatively, you can create a resource tool as it will provide the necessary scaffolding: 

`php artisan nova:resource-tool hemp/resource-table-row`

You can clear out the generated code dealing with routes and such. In the generated `nova-components/ResourceTableRow/resources/js/tool.js`, register your custom component with the provided Vue instance:

```js
import YourResourceTableRow from './YourResourceTableRow'

Nova.booting((Vue, router) => {
    Vue.component('resource-table-row', YourResourceTableRow)
})
```

Nova will serve this javascript file and during the boot process will override the built-in globally-registered Nova component with your own.

**Comment by endytc** [comment]
I have followed your steps @davidhemphill , but found this error
![image](https://user-images.githubusercontent.com/1209684/48032511-79343080-e18a-11e8-8780-b9f71f804118.png)

**Comment by DianaR19** [comment]
@davidhemphill Could you share a more detailed example, please?

**Comment by kjjdion** [comment]
Dude your proposal for us to override the built-in Vue component in order to facilitate model-specific functionality doesn't make a whole lot of sense.

I can't think of a single semi-complex app that wouldn't require non-bulk actions for one or more models.

**Comment by SerjoA** [comment]
@davidhemphill can you please help, i still didnt figure out how to use the 
computed fields in a table row to trigger an action i have added
i understand that i need to use the asHtml, but how do i trigger the action

**Comment by christianmarth** [comment]
Can we please have a more user friendly way to achieve this and also some more documentation as currently I'm scratching my head trying to figure out how to add a simple button to a resource

**Comment by SerjoA** [comment]
@christianmarth take a look at this: 
https://github.com/dillingham/nova-button/
but be aware, its without action modals, only backend events and some custom ones

**Comment by baldmike** [comment]
Closed in November (looks like several threads, all the same thing, closed), but still plenty of us PAID USERS 6 months later trying to figure out what should be out-of-the-box functionality.  I'm currently trying to hack this paid product by adding yet another package...  "Use actions" -- I did, and they make the user take several unnecessary, clunky, wordpress-esque steps - not why I paid for an admin "solution", and I'm baffled at how something so basic could be an afterthought.

**Comment by 0xbkt** [comment]
@baldmike Nova is not rock-solid.

**Comment by patricktsg** [comment]
Anyone else having trouble viewing their custom-index-toolbars in the latest version of Nova? They showed up fine before i upgraded now nothing.

**Comment by gbrits** [comment]
I managed to do this by overriding the 'action-selector' component as a new tool. (see nova/resources/js/components.js for the list of components you can overwrite).

You may have to adjust the mixin path in your tool, I copied the code over into my own tool & adjusted the `@/mixins/HandlesActions` path to be more old-school relative, aka: 
    import HandlesActions from '../../../../../nova/resources/js/mixins/HandlesActions'

And I also had to run `npm install --save laravel-nova` as it is a dependancy, but once you have the groundwork in place it works as @davidhemphill describes. I replaced the select with simple button repeat, and it worked:

    <button v-for="action in actions" @click="selectedActionKey = action.uriKey; determineActionStrategy()">
        {{ action.name }}
    </button>

**Comment by Poseidonas** [comment]
Hi,
In laravel nova resource index i want to add custom input fields(text form checkbox) to interact with table via submit button. The same operation with filters but i don want in table's right corner. i want them above the table.
Is also possible to hide defalt create button and form? 
how i set custom tool (vue component) but to interact with table index?[Example ][1]


  [1]: https://i.stack.imgur.com/xgfEt.png
